src/low-level/mime/mailmime_content.c

```diff
@@ -1754,9 +1767,11 @@ static int write_decoded_qp(MMAPString * mmapstr,
 
 #define WRITE_MAX_QP 512
 
-int mailmime_quoted_printable_body_parse(const char * message, size_t length,
+static int mailmime_quoted_printable_body_parse_impl(
```

Comment:
I liked in `mailmime_base64_body_parse_impl` how you made sure the behavior was changed only in case when partial == 1.
Could you have a similar treatment here?
for example, the new line 1815 is new and always executes.